dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -182,6 +190,9 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
             s"Event cannot mature in the past, got $maturationTime")
 
     for {
+      dbs <- eventDAO.findByEventName(eventName)
+      _ = require(dbs.isEmpty, s"Event name ($eventName) is already being used")
```

Comment:
Why don't we just enforce this at the database level?

Reply:
Because we have mulitple event dbs per event (one for each nonce)

Reply:
Can't we have an aggregate unique constraint between the `event_name` and `event_descriptor_tlv` or `announcement_signature`

https://www.sqlitetutorial.net/sqlite-unique-constraint/

>  UNIQUE(column_name1,column_name2,...)

I'm sure there is something similar for postgres

Reply:
After doing some research I don't think it is possible to have a constraint that enforces that only `event_name` can be paired with its one `announcement_signature`